client/ayon_core/plugins/publish/validate_asset_docs.py

```diff
@@ -37,3 +49,156 @@ def process(self, instance):
+        AYONPyblishPluginMixin
+    ):
+    """ Validate resolution values before updating an existing folder.
+    """
+
+    label = "Validate new folder resolution"
+    order = pyblish.api.ValidatorOrder
+    families = ["shot"]
+    actions = [RepairIgnoreResolution, RepairOverrideResolution]
+
+    @classmethod
+    def get_shot_data(self, hierarchy_context):
+        """ Retrieve matching entity and shot_data from hierarchy_context.
+        """
+        project_name = tuple(hierarchy_context.keys())[0]
+        entity_hub = EntityHub(project_name)
+        entity_data = {project_name: entity_hub.project_entity}
+        entity_to_inspect = [(entity_data, hierarchy_context)]
+
+        while entity_to_inspect:
+            entity_data, data = entity_to_inspect.pop(0)
+
+            for name, value in data.items():
+                entity = entity_data.get(name)
+                child = value.get("children", None)
+
+                if entity and child:
+                    entity_children = {
+                        chld.name: chld
+                        for chld in entity.children
+                    }
+                    entity_to_inspect.append((entity_children, child))
+
+                # Destination shot already exists return for validation.
+                elif (
+                    value.get("folder_type") == "Shot"
+                    and entity and not child
+                ):
+                    shot_data = value.get("attributes", {})
+                    return entity_hub, entity, shot_data
+
+        return None
+
+    def process(self, instance):
+        """ Validate existing shot resolution.
+        """
+        try:
+            hierarchy_context = instance.context.data["hierarchyContext"]
+
+        except KeyError:
+            self.log.debug("No hierarchy context defined for shot instance.")
+            return
+
+        validation_data = self.get_shot_data(hierarchy_context)
+        if not validation_data:
+            self.log.debug(
+                "Destination shot does not exist yet, "
+                "nothing to validate."
+            )
+            return
+
+        values = self.get_attr_values_from_data(instance.data)
+        _, entity, shot_data = validation_data
+
+        # Validate existing shot resolution is matching new one
+        # ask for confirmation instead of blind update, this prevents mistakes.
+        if values.get("updateExistingFolderResolution", True):
+            similar = True
+            for resolution_attrib in _RESOLUTION_ATTRIBS:
+                shot_value = shot_data.get(resolution_attrib)
+                entity_value = entity.attribs.get(resolution_attrib)
+                if shot_value and shot_value != entity_value:
+                    self.log.warning(
+                        "Resolution mismatch for shot."
+                        f"{resolution_attrib}={shot_value} but "
+                        f"already existing shot is set to {entity_value}."
```

Comment:
```suggestion
                        f" existing shot is set to {entity_value}."
```
I think shorter is better in these logs, if it's just as clear.